src/activity_handlers.cpp

```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
         return;
     }
 
+    bool has_table_nearby = false;
+    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 1 ) ) {
+        if( g->m.furn( pt ) == furn_id( "f_table" ) ) {
+            has_table_nearby = true;
+        }
+    }
+    // workshop butchery (full) prequisites
+    if( action == BUTCHER_FULL ) {
+            if( g->m.furn( u.pos() ) != furn_id( "f_butcher_rack" ) ){
```

Comment:
The same here. In case you're planning to add some portable variant of butchering rack, add a flag to every variant and check for it here.

Reply:
In light of current discussion its the best way to do this.